modules/backend/controllers/Files.php

```diff
@@ -51,14 +52,54 @@ public function thumb($code = null, $width = 100, $height = 100, $mode = 'auto',
         return Response::make(View::make('backend::404'), 404);
     }
 
+    /**
+     * Attempt to return a redirect to a temporary URL to the asset instead of streaming the asset - if supported
+     *
+     * @param System|Models\File $file
+     * @param string|null $path Optional, defaults to the getDiskPath() of the file
+     * @return string|null
+     */
+    protected static function getTemporaryUrl($file, $path = null)
+    {
+        // Get the disk and adapter used
+        $url = null;
+        $disk = $file->getDisk();
+        $adapter = $disk->getAdapter();
+        if (is_a($adapter, 'League\Flysystem\Cached\CachedAdapter')) {
+            $adapter = $adapter->getAdapter();
+        }
+
+        if (is_a($adapter, 'League\Flysystem\AwsS3v3\AwsS3Adapter') ||
```

Comment:
@LukeTowers Should this condition be changed to just check for the `getTemporaryUrl` method? The other two class checks are probably redundant.

Reply:
If you can verify that works, sure. I was mirroring https://github.com/laravel/framework/blob/5.5/src/Illuminate/Filesystem/FilesystemAdapter.php#L471-L479 which has the explicit checks. @bennothommo

Reply:
@LukeTowers Ah ok - it seems those checks are for using different methods though (`getAwsTemporaryUrl` and `getRackspaceTemporaryUrl`). If you've got everything feeding through to the one `getTemporaryUrl` method, then it would just be a case of checking for the existence of that method, I would think.

Reply:
I don't know if AWS & Rackspace adaptors have that method though, the laravel code doesn't call those methods on those adaptors.

Reply:
Looked into it more, they don't have those methods on the adaptor itself.

Reply:
Yep, my mistake, I see how it all works now. Your implementation is good. :)